packages/core/admin/ee/server/services/audit-logs.js

```diff
@@ -58,25 +61,38 @@ const createAuditLogsService = (strapi) => {
     };
   };
 
-  const handleEvent = (name, ...args) => {
-    if (!name) {
-      throw Error('Name is required');
-    }
-
+  const handleEvent = async (name, ...args) => {
     const processedEvent = processEvent(name, ...args);
 
     if (processedEvent) {
-      // TODO: save events here via provider
-      console.log('Saving event', processedEvent);
+      await provider.saveEvent(processedEvent);
     }
   };
 
   return {
-    register() {
+    async register() {
       this._eventHubUnsubscribe = strapi.eventHub.subscribe(handleEvent);
+      await provider.register({ strapi });
       return this;
     },
 
+    async findMany(query) {
+      const { results, pagination } = await provider.findMany(query);
+
+      const sanitizedResults = results.map((result) => {
+        const { user, ...rest } = result;
+        return {
+          ...rest,
+          user: getService('user').sanitizeUser(user),
```

Comment:
This currently throws an error when an audit log doesn't have an author (author isn't mandatory), I think we could use a ternary to either return null or the sanitized user